packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Models/PropertyDescription.cs

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Generator.CSharp.Input;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Generator.CSharp
+{
+    internal class PropertyDescription
+    {
+        private readonly InputModelProperty _property;
+        private readonly CSharpType _type;
+        private readonly SerializationFormat _serializationFormat;
+        private readonly bool _isReadOnly;
+
+        /// <summary>
+        /// The constructed description for the property.
+        /// </summary>
+        public FormattableString Description { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PropertyDescription"/> class and constructs the description for the property.
+        /// </summary>
+        /// <param name="property"> The property for which the description is being constructed.</param>
+        /// <param name="type">The CSharpType of the property.</param>
+        /// <param name="serializationFormat">The serialization format of the property.</param>
+        /// <param name="isPropertyReadOnly">Flag used to determine if the property <paramref name="property"/> is read-only.</param>
+        public PropertyDescription(InputModelProperty property, CSharpType type, SerializationFormat serializationFormat, bool isPropertyReadOnly)
```

Comment:
Not sure if there is value in having this as an instance type.  I believe the union description is a static method since there are a couple entry points into this type we want to have.

Lets convert this to an internal static type with 2 entry points one which takes InputModelProperty and one which takes union types.

Lets also call it PropertyDescriptionBuilder